cli/ops/net.rs

```diff
@@ -379,57 +363,36 @@ fn op_shutdown(
   Ok(json!({}))
 }
 
-#[allow(dead_code)]
 struct TcpListenerResource {
-  listener: TcpListener,
-  waker: Option<futures::task::AtomicWaker>,
-  local_addr: SocketAddr,
+  listener: AsyncRefCell<TcpListener>,
 }
 
-impl Drop for TcpListenerResource {
-  fn drop(&mut self) {
-    self.wake_task();
+impl Resource for TcpListenerResource {
+  fn name(&self) -> Cow<str> {
+    "tcpListener".into()
   }
 }
 
 impl TcpListenerResource {
-  /// Track the current task so future awaiting for connection
-  /// can be notified when listener is closed.
-  ///
-  /// Throws an error if another task is already tracked.
-  pub fn track_task(&mut self, cx: &Context) -> Result<(), AnyError> {
-    // Currently, we only allow tracking a single accept task for a listener.
-    // This might be changed in the future with multiple workers.
-    // Caveat: TcpListener by itself also only tracks an accept task at a time.
-    // See https://github.com/tokio-rs/tokio/issues/846#issuecomment-454208883
-    if self.waker.is_some() {
-      return Err(custom_error("Busy", "Another accept task is ongoing"));
-    }
-
-    let waker = futures::task::AtomicWaker::new();
-    waker.register(cx.waker());
-    self.waker.replace(waker);
-    Ok(())
+  fn try_borrow_mut(self: Rc<Self>) -> Option<AsyncMut<TcpListener>> {
+    RcRef::map(self, |r| &r.listener).try_borrow_mut()
   }
+}
 
-  /// Notifies a task when listener is closed so accept future can resolve.
-  pub fn wake_task(&mut self) {
-    if let Some(waker) = self.waker.as_ref() {
-      waker.wake();
-    }
-  }
+struct UdpSocketResource {
+  socket: AsyncRefCell<UdpSocket>,
```

Comment:
This resource needs to be split into a readable and a writable end, just like `TcpStream` in the deno_core benchmarks.
Otherwise it can't send packets while it is also receiving.

Reply:
It looks like both send_to() and recv_from() can work on a `&UdpSocket`, so rather than splitting it we could just use borrow() instead of borrow_mut().

Reply:
Continued in https://github.com/denoland/deno/pull/8512#discussion_r543683783.